app/src/main/java/org/pesmypetcare/mypetcare/services/VetVisitsAdapter.java

```diff
@@ -0,0 +1,88 @@
+                visit.getBody().getReason()));
+        }
+        return appVisits;
+    }
+
+    @Override
+    public void createVetVisit(User user, Pet pet, VetVisit vetVisit) throws ExecutionException, InterruptedException {
+        String accessToken = user.getToken();
+        String owner = user.getUsername();
+        String petName = pet.getName();
+
+        VetVisitData visitData = new VetVisitData(vetVisit.getReason(), vetVisit.getAddress());
+        org.pesmypetcare.usermanagerlib.datacontainers.VetVisit libraryVisit =
+            new org.pesmypetcare.usermanagerlib.datacontainers.VetVisit(vetVisit.getVisitDate().toString(), visitData);
+
+        ServiceLocator.getInstance().getPetManagerClient().addFieldCollectionElement(accessToken, owner, petName,
+            PetData.VET_VISITS, libraryVisit.getKey(), libraryVisit.getBodyAsMap());
+    }
+
+    @Override
+    public void deleteVetVisit(User user, Pet pet, VetVisit vetVisit) throws ExecutionException, InterruptedException {
+        String accessToken = user.getToken();
+        String owner = user.getUsername();
+        String petName = pet.getName();
+        VetVisitData visitData = new VetVisitData(vetVisit.getReason(), vetVisit.getAddress());
+        org.pesmypetcare.usermanagerlib.datacontainers.VetVisit libraryVisit =
+            new org.pesmypetcare.usermanagerlib.datacontainers.VetVisit(vetVisit.getVisitDate().toString(), visitData);
+        ServiceLocator.getInstance().getPetManagerClient().deleteFieldCollectionElement(accessToken, owner, petName,
+            PetData.VET_VISITS, libraryVisit.getKey());
+    }
+
+    @Override
+    public void updateVetVisitKey(User user, Pet pet, String newDate, DateTime visitDate) throws ExecutionException, InterruptedException {
+        String accessToken = user.getToken();
+        String owner = user.getUsername();
+        String petName = pet.getName();
+        VetVisitData vetVisitData = ServiceLocator.getInstance().getPetCollectionsManagerClient().getVetVisit
+            (accessToken, owner, petName, visitDate.toString());
+        org.pesmypetcare.usermanagerlib.datacontainers.VetVisit oldLibraryVisit =
+            new org.pesmypetcare.usermanagerlib.datacontainers.VetVisit(visitDate.toString(), vetVisitData);
+        org.pesmypetcare.usermanagerlib.datacontainers.VetVisit newLibraryVisit =
+            new org.pesmypetcare.usermanagerlib.datacontainers.VetVisit(newDate, vetVisitData);
+        ServiceLocator.getInstance().getPetManagerClient().deleteFieldCollectionElement(accessToken, owner, petName,
+            PetData.VET_VISITS, oldLibraryVisit.getKey());
+        ServiceLocator.getInstance().getPetManagerClient().addFieldCollectionElement(accessToken, owner, petName,
+            PetData.VET_VISITS, newLibraryVisit.getKey(), newLibraryVisit.getBodyAsMap());
+
+    }
+
+    @Override
+    public void updateVetVisitBody(User user, Pet pet, VetVisit vetVisit) throws ExecutionException, InterruptedException {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line is longer than 120 characters (found 123).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5493224)